Nerf the DUMP() macro to elide the dumped value

Drake builds using EIGEN_NO_IO, so Conex can't use it either.

--- conex/debug_macros.h
+++ conex/debug_macros.h
@@ -8,7 +8,7 @@
 #define DUMP(x)                                              \
   std::cerr << __FILE__ << " line " << __LINE__ << std::endl \
             << #x ":" << std::endl                           \
-            << x << std::endl;
+            << ((void)(x), "(not printed)") << std::endl;
 
 #ifndef CONEX_VERBOSE
 #define CONEX_VERBOSE 1
